### PR TITLE
ref(newsletter): Various cleanup and fixes

### PR DESCRIPTION
**Pull request by dcramer**
- Pass correct list_id for default newsletter subscription
- Fill in function signatures on Newsletter base
- Add DummyNewsletter for tests
- Utilize Newsletter.is_enabled from service
- Remove sentry.app.newsletter (in favor of sentry.newsletter)

### CODE REVIEW

File: src/sentry/api/endpoints/user_emails.py [hunk 1]
```diff
@@ -60,10 +60,11 @@ def add_email(email, user, subscribe_newsletter=False):
 
         # Update newsletter subscription and mark as unverified
         if subscribe_newsletter:
-            newsletter.update_subscription(user=user,
-                                           verified=False,
-                                           list_id=1,
-                                           )
+            newsletter.update_subscription(
```
Comment (dcramer) [inline src/sentry/api/endpoints/user_emails.py]: should this be create_or_update?

Comment (mikeclarke) [reply src/sentry/api/endpoints/user_emails.py]: `newsletter.create_or_update_subscription(...)` ya

**Review by mikeclarke — APPROVED** [review]

**Comment by dcramer** [comment]
This should now fail until https://github.com/getsentry/sentry/pull/7981 lands

**Comment by dcramer** [comment]
@mikeclarke this is quite significantly different than before. Lots of various cleanup/tests, and a full mock (DummyNewsletter) for testing against.

**Comment by dcramer** [comment]
I think this might resolve a bug with the typed field in master (that I couldn't test at the time). Its green locally so going to merge this to avoid any actual concerns.

**Pull request merged**